src/overlays/fbdemos/ovl_fbdemo_wipe4/z_fbdemo_wipe4.c

```diff
@@ -1,11 +1,22 @@
-/*
- * File: z_fbdemo_wipe4.c
+/**
+ * @file z_fbdemo_wipe4.c
  * Overlay: ovl_fbdemo_wipe4
  * Description:
+ *
+ * @note The bug in TransitionWipe4_Draw() makes this transition do nothing apart from increase its timer until it is
+ * finished.
  */
 
 #include "global.h"
 #include "z_fbdemo_wipe4.h"
+#include "sys_cfb.h"
+
+#define THIS ((TransitionWipe4*)thisx)
+
+// TODO: determine how generic these macros are. Requires rest of the fbdemos decompiled.
+#define TW4_SET_PARAMS (1 << 7)
+#define TW4_GET_COLORTYPE(type) (((type) >> 1) & 3)
+#define TW4_GET_SPEEDTYPE(type) ((type)&1)
```

Comment:
Could you make these similar with `fbdemo_wipe3` (or possibly the other way around and use TW3 as the prefix over there which actually seems better. (Could you also add parens in the wipe3 macros)